components/calendar/calendar.js

```diff
@@ -0,0 +1,488 @@
+			const weekHtml = week.map((day) => {
+				const classes = {
+					'd2l-calendar-date': true,
+					'd2l-calendar-date-selected': this.selectedValue ? checkIfDatesEqual(day, parseDate(this.selectedValue)) : false,
+					'd2l-calendar-date-today': checkIfDatesEqual(day, this._today)
+				};
+				const focused = checkIfDatesEqual(day, this._focusDate);
+				return html`
+					<td>
+						<div
+							@click="${this._onDateSelected}"
+							@keydown="${this._onKeyDown}"
+							class=${classMap(classes)}
+							data-date=${day.getDate()}
+							data-month=${day.getMonth()}
+							data-year=${day.getFullYear()}
+							tabindex=${focused ? '0' : '-1'}>
+							${day.getDate()}
+						</div>
+					</td>`;
+			});
+
+			return html`<tr>${weekHtml}</tr>`;
+		});
+		const heading = `${descriptor.calendar.months.long[this._shownMonth]} ${this._shownYear}`;
+		return html`
+			<div aria-labelledby="${this._labelId}" class="d2l-calendar">
+				<table summary="${this.summary}" role="grid">
+					<thead>
+						<tr class="d2l-calendar-title">
+							<td>
+								<d2l-button-icon
+									@click="${this._onPrevMonthButtonClick}"
+									text="${this._computeText(getPrevMonth(this._shownMonth))}"
+									icon="tier1:chevron-left">
+								</d2l-button-icon>
+							</td>
+							<th colspan="5">
+								<h2 class="d2l-heading-4" aria-live="polite" id="${this._labelId}">${heading}</h2>
+							</th>
+							<td>
+								<d2l-button-icon
+									@click="${this._onNextMonthButtonClick}"
+									text="${this._computeText(getNextMonth(this._shownMonth))}"
+									icon="tier1:chevron-right">
+								</d2l-button-icon>
+							</td>
+						</tr>
+						<tr>${weekdayHeaders}</tr>
+					</thead>
+					<tbody>
+						${dayRows}
+					</tbody>
+				</table>
+				<slot></slot>
+			</div>
+		`;
+	}
+
+	focus() {
+		const firstButton = this.shadowRoot.querySelector('d2l-button-icon');
+		if (firstButton) firstButton.focus();
+	}
+
+	_computeText(month) {
+		return this.localize('show', {month: descriptor.calendar.months.long[month]});
+	}
+
+	_focusDateAddFocus() {
+		const date = this._getFocusDateElement();
+		if (date && this._keyboardTriggeredMonthChange) {
+			date.focus();
+			this._keyboardTriggeredMonthChange = false;
+		}
+	}
+
+	_getFocusDateElement() {
+		return this.shadowRoot.querySelector(`div[data-date="${this._focusDate.getDate()}"][data-month="${this._focusDate.getMonth()}"][data-year="${this._focusDate.getFullYear()}"]`);
+	}
+
+	_onDateSelected(e) {
+		const selectedDate = e.composedPath()[0];
+		const year = selectedDate.getAttribute('data-year');
+		let month = selectedDate.getAttribute('data-month');
+		let date = selectedDate.getAttribute('data-date');
+		this._focusDate = new Date(year, month, date);
+		month = (parseInt(month) + 1).toString();
+		if (month.length < 2) {
+			month = `0${month}`;
+		}
+		if (date.length < 2) {
+			date = `0${date}`;
+		}
+
+		this.selectedValue = [year, month, date].join('-');
```

Comment:
Nit: Could be handy to centralize this in a `formatSelectedValue()` that takes a `Date` (or a month/year/day). Basically the opposite if your `parseDate`.